### PR TITLE
Soft test set

### PR DESCRIPTION
**Pull request by jdiaz4302**
Very small PR that I'm pushing through without review.

### What's different?

I added a few lines of code to implement the "soft" test sets discussed in https://github.com/jdiaz4302/lake_ice_prediction/pull/11

These are additional data that were discarded and thus not used for weight training or hyperparameter selection, but they are more similar to the training/validation set than the true test set (which has different lakes in different years). There ends up being two types of soft test data:

1. Data for lakes that were used during training but for years that were withheld for testing; I later abbreviate this as "TYTrL" (**T**est **Y**ears, **Tr**ain **L**akes).
3. Data for lakes that were withheld for testing but for years that were used during training; I later abbreviate this as "TrYTL" (**Tr**ain **Y**ears, **T**est **L**akes).

It was quickly verified (via initially saving to different file paths) that this does not change the existing training, validation or test sets.

### CODE REVIEW

File: 01_process/src/C_TrainTestSplits.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ try_n_test_partitions = 10000
 ```python
 train_data_fpath = out_dir + 'train_data.npz'
 valid_data_fpath = out_dir + 'valid_data.npz'
+soft_test_data_fpath = out_dir + 'soft_test_data.npz'
```
Comment (jdiaz4302) [inline 01_process/src/C_TrainTestSplits.md:75]: new fpath

File: 01_process/src/C_TrainTestSplits.md [hunk 2]
```diff
@@ -380,9 +381,21 @@ def save_partition_data(indices, fpath):
     np.savez_compressed(fpath, **part_data)
 ```
 
+```python
+soft_test_indices = []
+for i in df.index:
+    if (i not in train_df.index) and (i not in valid_df.index) and (i not in test_df.index):
+        soft_test_indices.append(i)
```
Comment (jdiaz4302) [inline 01_process/src/C_TrainTestSplits.md:388]: Find the discarded data

File: 01_process/src/C_TrainTestSplits.md [hunk 3]
```diff
@@ -380,9 +381,21 @@ def save_partition_data(indices, fpath):
...
+```
+
+```python
+assert (len(soft_test_indices) + train_df.shape[0] + valid_df.shape[0] + test_df.shape[0]) == df.shape[0]
+```
+
 ```python
 save_partition_data(train_df.index, train_data_fpath)
 save_partition_data(valid_df.index, valid_data_fpath)
+save_partition_data(soft_test_indices, soft_test_data_fpath)
```
Comment (jdiaz4302) [inline 01_process/src/C_TrainTestSplits.md:398]: Save the discarded data
